Samples/DirectMLNpuInference/main.cpp

```diff
@@ -30,36 +65,33 @@ void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** comma
             dxcoreCreateAdapterFactory(IID_PPV_ARGS(&factory));
         }
     }
-    // Create the DXCore Adapter
+
+    // Create the DXCore Adapter, for the purposes of the sample we look for (!GRAPHICS && (GENERIC_ML || CORE_COMPUTE))
     ComPtr<IDXCoreAdapter> adapter;
+    ComPtr<IDXCoreAdapterList> adapterList;
     if (factory)
     {
-        const GUID dxGUIDs[] = { DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
-        ComPtr<IDXCoreAdapterList> adapterList;
-        THROW_IF_FAILED(factory->CreateAdapterList(ARRAYSIZE(dxGUIDs), dxGUIDs, IID_PPV_ARGS(&adapterList)));
-        for (uint32_t i = 0, adapterCount = adapterList->GetAdapterCount(); i < adapterCount; i++)
-        {
-            ComPtr<IDXCoreAdapter> currentGpuAdapter;
-            THROW_IF_FAILED(adapterList->GetAdapter(static_cast<uint32_t>(i), IID_PPV_ARGS(&currentGpuAdapter)));
+        THROW_IF_FAILED(factory->CreateAdapterList(1, &DXCORE_ADAPTER_ATTRIBUTE_D3D12_GENERIC_ML, IID_PPV_ARGS(&adapterList)));
 
-            if (!forceComputeOnlyDevice && !forceGenericMLDevice)
-            {
-                // No device restrictions
-                adapter = std::move(currentGpuAdapter);
-                break;
-            }
-            else if (forceComputeOnlyDevice && currentGpuAdapter->IsAttributeSupported(DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE))
-            {
-                adapter = std::move(currentGpuAdapter);
-                break;
-            }
-            else if (forceGenericMLDevice && currentGpuAdapter->IsAttributeSupported(DXCORE_ADAPTER_ATTRIBUTE_D3D12_GENERIC_ML))
-            {
-                adapter = std::move(currentGpuAdapter);
-                break;
-            }
+        if (adapterList->GetAdapterCount() > 0)
+        {
+            GetNonGraphicsAdapter(adapterList, adapter);
+        }
+        
+        if (!adapter)
+        {
+            THROW_IF_FAILED(factory->CreateAdapterList(1, &DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE, IID_PPV_ARGS(&adapterList)));
+            GetNonGraphicsAdapter(adapterList, adapter);
         }
     }
+
+    if (adapter)
+    {
+        std::string adapterName;
+        TryGetProperty(adapter, DXCoreAdapterProperty::DriverDescription, adapterName);
```

Comment:
TryGetProperty returns a boolean, but the return value is not checked. So `printf("Successfully found adapter %s\n", adapterName.c_str());` always purports success, even if it failed (which will be a confused diagnostic).

Reply:
Good catch! Resolved, thanks!